lightning/src/ln/channel.rs

```diff
@@ -2081,7 +2093,12 @@ impl<Signer: Sign> Channel<Signer> {
 		} else if their_features.supports_channel_type() {
 			// Assume they've accepted the channel type as they said they understand it.
 		} else {
-			self.channel_type = ChannelTypeFeatures::from_counterparty_init(&their_features)
+			self.channel_type = ChannelTypeFeatures::from_counterparty_init(&their_features);
+			self.channel_transaction_parameters.opt_anchors = if self.channel_type.supports_anchors_zero_fee_htlc_tx() {
+					Some(())
+			} else {
+					None
```

Comment:
I think here we can have a situation where our counterparty assumes we have another channel type like static_remote_key or simple anchor, and that could lead to a signature wrong on the first channel commitment transaction exchanged, am i correct ? In case of feature negotiation failure we could also fail the channel opening and give back an error message, I guess.